docs/source/working_with_aiida/caching.rst

```diff
@@ -65,88 +115,76 @@ This means that caching is enabled for ``TemplatereplacerCalculation`` and ``Str
+2. Clear the hashes of all nodes that are considered identical to this node:
 
     .. code:: python
 
         for n in node.get_all_same_nodes():
             n.clear_hash()
-3. Run your calculation again. Now it should not use caching.
-
-If you instead think that there is a bug in the AiiDA implementation, please open an issue (with enough information to be able to reproduce the error, otherwise it is hard for us to help you) in the AiiDA GitHub repository: https://github.com/aiidateam/aiida-core/issues/new.
-
-.. _caching_provenance:
+3. Run your calculation again. The node in question should no longer be reused.
 
-Caching and the Provenance Graph
---------------------------------
 
-The goal of the caching mechanism is to speed up AiiDA calculations by re-using duplicate calculations. However, the resulting provenance graph should be exactly the same as if caching was disabled. This has important consequences on the kind of caching operations that are possible.
-
-The provenance graph consists of nodes describing data, calculations and workchains, and links describing the relationship between these nodes. We have seen that the hash of a node is used to determine whether two nodes are equivalent. To successfully use a cached node however, we also need to know how the new node should be linked to its parents and children.
-
-In the case of a plain data node, this is simple: Copying a data node from an equivalent node should not change its links, so we just need to preserve the links which this new node already has.
-
-For calculations, the situation is a bit more complex: The node can have inputs and creates new data nodes as outputs. Again, the new node needs to keep its existing links. For the outputs, the calculation needs to create a copy of each node and link these as its outputs. This makes it look as if the calculation had produced these outputs itself, without caching.
-
-Finally, workchains can create links not only to nodes which they create themselves, but also to nodes created by a calculation that they called, or even their ancestors. This is where caching becomes impossible. Consider the following example (using workfunctions for simplicity):
-
-.. code:: python
+.. _caching_limitations:
 
-    from aiida.engine import workfunction
-    from aiida.orm import Int
+Limitations
+-----------
 
-    @workfunction
-    def select(a, b):
-        return b
+#. While caching saves unnecessary computations, the current implementation does not yet save disk space:
+   The output nodes of the cached calculation are full copies of the original outputs.
+   The plan is to add data deduplication as a global feature at the repository and database level (independent of caching).
 
-    d = Int(1)
-    r1 = select(d, d)
-    r2 = select(Int(1), Int(1))
+#. Workflow nodes are not cached. As explained below, in the current design this follows from the requirement that the provenance graph be independent of whether caching is enabled or not.
```

Comment:
"As explained below" this is probably remnant from before reorganization. Does not make sense now.

Reply:
it was meant to refer to the itemized list below, but it is not really necessary.
I've removed it.